snooty/language_server.py

```diff
@@ -154,19 +153,25 @@ def on_progress(self, progress: int, total: int, message: str) -> None:
     def on_diagnostics(self, path: FileId, diagnostics: List[types.Diagnostic]) -> None:
         self.server.set_diagnostics(path, diagnostics)
 
-    def on_update(self, prefix: List[str], page_id: FileId, page: types.Page) -> None:
+    def on_update(
+        self,
+        prefix: List[str],
+        build_identifiers: Dict[str, Optional[str]],
```

Comment:
OH one additional thought: it may be worth having a `BuildIdentifierSet = Dict[str, Optional[str]]` in types.py to make these type signatures a little more semantic?